pex/bin/pex.py

```diff
@@ -189,6 +190,15 @@ def configure_clp_pex_resolution(parser):
     callback=process_transitive,
     help='Whether to transitively resolve requirements. Default: True')
 
+  group.add_option(
+    '-j', '--parallel',
+    metavar='JOBS',
+    dest='max_parallel_jobs',
+    type=int,
+    default=DEFAULT_MAX_JOBS,
+    help='The maximum number of parallel jobs to use when resolving, building and installing '
+         'distributions. [Default: %default]')
```

Comment:
- "Increase the maximum number of parallel jobs to potentially improve the latency of the pex creation process at the expense of other processes on your system."

That would be a concise snippet that conveys why in the cases I can think of. The decrease why is the inverse so I leave out. Let me know if you find that helpful to add to the help string after consideringprior art in directly relevant tools:
```
$ make --help 2>&1 | grep jobs
  -j [N], --jobs[=N]          Allow N jobs at once; infinite jobs with no arg.
                              Don't start multiple jobs unless load is below N.
                              Synchronize output of parallel jobs by TYPE.
$ ninja --help 2>&1 | grep jobs
  -j N     run N jobs in parallel (0 means infinity) [default=10 on this system]
  -k N     keep going until N jobs fail (0 means infinity) [default=1]
  -l N     do not start new jobs if the load average is greater than N
$ clang --help 2>&1 | grep jobs
  -flto-jobs=<value>      Controls the backend parallelism of -flto=thin (default of 0 means the number of threads will be derived from the number of CPUs detected)
```

In other words, by implication, the why is widespread knowledge. That's not a great answer, since it still begs the question of how it became widespread since no tools I know of explain the why. The reason I suspect the why is not needed, is the tool is used by programmers and they generally understand:
1. My machine is melting down -> decrease parallelism.
2. This thing is slow -> increase.

Reply:
> "Increase the maximum number of parallel jobs to potentially improve the latency of the pex creation process at the expense of other processes on your system."

This is great! Very helpful. 

Rather than relying on the assumption that Pex users already know the why behind this option, this small snippet makes our tool more accessible for beginners, which is a big win in my book.

Reply:
I don't agree here, but done.